x/concentrated-liquidity/incentives.go

```diff
@@ -851,15 +851,24 @@ func (k Keeper) CreateIncentive(ctx sdk.Context, poolId uint64, sender sdk.AccAd
 		return types.IncentiveRecord{}, types.NonPositiveEmissionRateError{PoolId: poolId, EmissionRate: emissionRate}
 	}
 
-	// Ensure min uptime is one of the supported periods
+	// Ensure min uptime is one of the authorized uptimes.
+	// Note that this is distinct from the supported uptimes – while we set up pools and positions to
+	// accommodate all supported uptimes, we only allow incentives to be created for uptimes that are
+	// authorized by governance.
+	authorizedUptimes := k.GetParams(ctx).AuthorizedUptimes
+	osmoutils.SortSlice(authorizedUptimes)
```

Comment:
What is the purpose of sorting?

Reply:
My assumption is that uptimes can technically be stored in non-increasing order, which I believe we might have some assumptions leaning on (even if just in tests). This was intended to be a low-cost guardrail to ensure that ordering is not an issue. Do you see an issue with this reasoning?